### PR TITLE
build: Update build.sh to run fully manual build

### PR DESCRIPTION
**Pull request by nadiasvertex**
Also, add ubuntu-16.10-x64 to the list of architectures to build for.

### CODE REVIEW

File: build.sh [hunk 1]
```diff
@@ -1,8 +1,27 @@
...
+
+   pushd src/TypeCatalogGen
+   dotnet run ../Microsoft.PowerShell.CoreCLR.AssemblyLoadContext/CorePsTypeCatalog.cs powershell.inc
+   popd
+
+   pushd src/libpsl-native
+   cmake -DCMAKE_BUILD_TYPE=Debug .
+   make -j
+   make test
+   popd
```
Comment (vors) [inline build.sh]: The biggest step is missing: `dotnet build` from `src/powershell-unix`

File: build.sh [hunk 2]
```diff
@@ -1,8 +1,27 @@
 #!/usr/bin/env bash
+set -e
 
 if hash powershell 2>/dev/null; then
     echo 'Continuing with `powershell -noprofile -c Start-PSBuild`'
     powershell -noprofile -c "Import-Module ./build.psm1; Start-PSBuild"
 else
-    echo 'No `powershell`, see docs/building/linux.md or macos.md to build PowerShell!'
+   echo 'Continuing with full manual build'
```
Comment (vors) [inline build.sh]: Maybe a separate script (i.e. `build-new-platform.sh` or something) would be better. The reason is the build system is already pretty complicated. Putting more logic into the main build path would increase the probability of future errors and time for trouble-shooting.

**Review by vors — CHANGES_REQUESTED** [review]

File: build.sh [hunk 3]
```diff
@@ -1,8 +1,10 @@
...
+   ./build-from-scratch.sh
```
Comment (vors) [inline build.sh]: Again, I don't think it should be run automatically.
The error message should suggest to use this script, but not run it.

File: build-from-scratch.sh [hunk 4]
```diff
@@ -0,0 +1,26 @@
+#!/usr/bin/env bash
```
Comment (vors) [inline build-from-scratch.sh]: I don't think `build-from-scratch` is a good name. `build.sh` also build everything from scratch.
Maybe `build-without-powershell.sh`? I think name should emphasise what's different about this build script.

**Review by vors — CHANGES_REQUESTED** [review]
When onboarding new unix distros, it's desirable to have docker build as well https://github.com/PowerShell/PowerShell/tree/master/docker 

For example see https://github.com/PowerShell/PowerShell/pull/2738/files#diff-7f8105a00850a7f7fea1de45bca96f8e

File: build.sh (hunk 3 above)
Comment (nadiasvertex) [inline build.sh]: Why? This makes no sense. The user has typed "build.sh". They expect the build to run. The script has proved to itself that powershell is not available. So now it should fail and tell the user what to do next? That seems to be the height of silly.

File: build-from-scratch.sh (hunk 4 above)
Comment (nadiasvertex) [inline build-from-scratch.sh]: Okay.

**Comment by nadiasvertex** [comment]
That sounds great. I will make th or changes.

**Comment by andschwa** [comment]
I like this change a lot 😄

**Comment by SteveL-MSFT** [comment]
@nadiasvertex your build issue with #2735 complaining about Microsoft.CodeAnalysis.dll is due to `src/TypeCatalogParser/Main.cs` the call `x.Name.ToLower()` making the filenames in lowercase and thus later on not matching the case-sensitive paths.  Not sure why this was there explicitly, but removing that call will get you past this.

**Comment by andschwa** [comment]
@SteveL-MSFT `x.Name.ToLower()` is there explicitly because of .NET CLI's behavior, per #2162. We verified this was intended change on their end in https://github.com/dotnet/cli/issues/4141.

**Comment by SteveL-MSFT** [comment]
@andschwa well, this is awkward since dotnet restore on my Ubuntu 16.10 system has packages with Pascal casing...

**Comment by andschwa** [comment]
@SteveL-MSFT file a bug with .NET CLI, see if they purposefully changed it back 😄

**Comment by SteveL-MSFT** [comment]
@andschwa I think the difference is a breaking change they made with preview3 vs preview2.  On my Windows box and Ubuntu16.10, I have preview2 (as far as I can tell, preview3 isn't available on 16.10).  This PR may have to wait until we have preview3.

**Comment by andschwa** [comment]
@SteveL-MSFT @nadiasvertex:

I installed the exact build that `Start-PSBootstrap` is currently pinned to on an Ubuntu 16.10 box, did a `dotnet restore`, and confirmed that these packages are *not* restored lower-cased:

```sh
docker run -it ubuntu:16.10 bash
apt-get update
apt-get install libunwind8 libicu57 libcurl3 git curl
curl -sO https://raw.githubusercontent.com/dotnet/cli/v1.0.0-preview2-1-3177/scripts/obtain/dotnet-install.sh
bash dotnet-install.sh -c preview -v 1.0.0-preview2-1-00317
git clone https://github.com/PowerShell/PowerShell.git
cd PowerShell/
~/.dotnet/dotnet --version
~/.dotnet/dotnet restore
ls ~/.nuget/packages/
```
And got:
```
1.0.0-preview2-1-003177.dotnetSentinel
Libuv
Microsoft.Build
Microsoft.Build.Framework
Microsoft.Build.Utilities.Core
Microsoft.CSharp
Microsoft.CodeAnalysis.Analyzers
Microsoft.CodeAnalysis.CSharp
Microsoft.CodeAnalysis.Common
Microsoft.CodeAnalysis.VisualBasic
Microsoft.DiaSymReader
Microsoft.DiaSymReader.Native
Microsoft.DotNet.Cli.Utils
...
```

So I dug a bit further, and in 32a8601f926b164bbdf74ad1b6c4c8ef927fe1a8 the `ToLower()` was removed because:

> .NET Core 1.1 ships with an older dotnet/cli than has currently been used, so we revert to use case-sensitive directory names for dependencies.

This commit was made December 8, so I'd recommend @nadiasvertex you pull PowerShell and get the latest changes, as the bug you hit appears fixed.

**Comment by andschwa** [comment]
(This corresponds to @SteveL-MSFT finding the change is in preview3, but not preview2; and I didn't realize we'd reverted the tooling back that far.)

**Pull request merged**